modules/apps/document-library/document-library-preview-pdf/src/main/java/com/liferay/document/library/preview/pdf/internal/configuration/admin/service/PDFPreviewManagedServiceFactory.java

```diff
@@ -79,6 +83,33 @@ public long getSystemMaxNumberPages() {
 		return _systemPDFPreviewConfiguration.maxNumberPages();
 	}
 
+	public void updateCompanyPDFPreview(long companyId, long maxNumberPages)
+		throws Exception {
+
+		Dictionary<String, Object> properties = null;
+
+		Configuration configuration = _getScopedConfiguration(
+			ExtendedObjectClassDefinition.Scope.COMPANY, companyId);
+
+		if (configuration == null) {
+			configuration = _configurationAdmin.createFactoryConfiguration(
+				PDFPreviewConfiguration.class.getName() + ".scoped",
+				StringPool.QUESTION);
+
+			properties = HashMapDictionaryBuilder.<String, Object>put(
+				ExtendedObjectClassDefinition.Scope.COMPANY.getPropertyKey(),
+				companyId
+			).build();
+		}
+		else {
+			properties = configuration.getProperties();
+		}
+
+		properties.put("maxNumberPages", maxNumberPages);
+
+		configuration.update(properties);
+	}
+
```

Comment:
This method is almost identical to the group version below. Can you extract the common logic and pass the ExtendedObjectClassDefinition.Scope as argument?

Reply:
good catch!!!